### ISSUE TITLE
Add a way to export class types without using ES6 class syntax

### ISSUE THREAD

**Issue by spicyj** [facebook/flow#23]
It doesn't seem possible to declare classes by writing out the constructor and prototype manually that Flow will properly recognize, along the lines of

```
function Foo(x: number) {
  this.x = x;
}
Foo.prototype.squared = function(): number {
  return this.x * this.x;
}
```

Is this off the table completely or is there a way that Flow can be made to recognize this sort of declaration?

**Comment by Raynos** [comment]
:+1:

It's incredibly hard to declare the type of a constructor function. I could not find any way to do it all; I tried everything from `declare class` to `declare function` to `declare var`

**Comment by samwgoldman** [comment]
Flow seems to understand @spicyj's original example now without any errors. Is there anything else remaining to be done here?

**Comment by Raynos** [comment]
@samwgoldman 

can we have some examples of how to write a ES5 constructor and then how to write an declare block that declares it's type (preferably in one of those block comments that flow supports).

The use case here is to annotate a complex existing constructor with many methods using the unobtrusive block comment workflow.

**Comment by samwgoldman** [comment]
@Raynos It would be easier to start with what you have tried that didn't work. Can you share some code?

**Comment by Raynos** [comment]
@samwgoldman I can try and reproduce it; I worked on it a few weeks ago.

Let me update the thread with a breaking example tonight.

**Comment by samwgoldman** [comment]
@avikchaudhuri seems to have addressed @spicyj's original issue in 618186ad5ceb66cf7bf009b2591b74ed885d69bd. I think this can be closed.

**Comment by spicyj** [comment]
Yes, thanks!
